src/SudoSu.php

```diff
@@ -23,15 +23,27 @@ public function __construct(Application $app, AuthManager $auth, SessionManager
         $this->session = $session;
     }
 
-    public function loginAsUser($userId, $currentUserId)
+    /**
+     * Log the sudosu user in and preserve the original user id if exists
+     *
+     * @param string $sudosuUserId
+     * @param string|null $originalUserId
+     * @return void
+     */
+    public function loginAsUser($sudosuUserId, $originalUserId)
     {
         $this->session->put('sudosu.has_sudoed', true);
-        $this->session->put($this->sessionKey, $currentUserId);
+        $this->session->put($this->sessionKey, $originalUserId);
 
-        $this->auth->loginUsingId($userId);
+        $this->auth->loginUsingId($sudosuUserId);
```

Comment:
I agree that explicit is always better than implicit (and I've already read Clean Code!), but I think in this case it's pretty explicit (in my opinion):

```php
function loginAsUser($userId, $originalUserId)
```

Does that cause confusion for you? Again, I'm not disputing, just wondering because it reads pretty expressively for me.

If anything (for me at least), naming it `$sudosuUserId` would cause me more confusion because I'd assume that a `sudosuUserId` is different to a regular user ID. Maybe that's just me though?

I'd be interested to get another person's thoughts on this too. /cc @krve 

Either way, thanks for the PR @Te7a-Houdini, I love when people contribute like this :) Appreciate it! 👍